devices/Ft6xx6x/samples/Program.cs

```diff
@@ -1,55 +1,226 @@
+                    DiscrimitateButtons(point.X, point.Y);
+                }
+                else if (points == 2)
+                {
+                    var dp = sensor.GetDoublePoints();
+                    Debug.WriteLine($"ID: {dp.Point1.TouchId}, X: {dp.Point1.X}, Y: {dp.Point1.Y}, Weight: {dp.Point1.Weigth}, Misc: {dp.Point1.Miscelaneous}, Event: {dp.Point1.Event}");
+                    Debug.WriteLine($"ID: {dp.Point2.TouchId}, X: {dp.Point2.X}, Y: {dp.Point2.Y}, Weight: {dp.Point2.Weigth}, Misc: {dp.Point2.Miscelaneous}, Event: {dp.Point1.Event}");
+                }
+            }
+
+            void DiscrimitateButtons(int x, int y)
+            {
+                const double pxPerMm = 6.90415;
+                //r in milimeters is the radious of sensitivity to detect any of the given 3 buttons
+                const double rMm = 3.5;
+                const int x0L = 83;
+                const int x0M = 182;
+                const int x0R = 271;
+                const int y0 = 263;
+
+                TouchEventArgs touchEventArgs = new TouchEventArgs();
+
+                //WARNING: Next assigment should be EventCategory.Touch but Touch Category not yet defined on main/develop
+                touchEventArgs.eventCategory = EventCategory.Unknown;
+                touchEventArgs.X1 = x;
+                touchEventArgs.Y1 = y;
+
+                //r^2 in pixels^2
+                int r2 = (int)((rMm * pxPerMm) * (rMm * pxPerMm));
+
+                int tmpL = 0;
+                int tmpM = 0;
+                int tmpR = 0;
+                int tmpY = 0;
+
+                tmpY = (y - y0) * (y - y0);
+                tmpL = (x - x0L) * (x - x0L);
+                tmpL += tmpY;
+                tmpM = (x - x0M) * (x - x0M);
+                tmpM += tmpY;
+                tmpR = (x - x0R) * (x - x0R);
+                tmpR += tmpY;
+
+                if (tmpL <= r2)
+                {
+                    touchEventArgs.eventSubCategory = (int)TouchEventSubcategory.LeftButton;
+                }
+                else if (tmpM <= r2)
+                {
+                    touchEventArgs.eventSubCategory = (int) TouchEventSubcategory.MiddleButton;
+                }
+                else if (tmpR <= r2)
+                {
+                    touchEventArgs.eventSubCategory = (int) TouchEventSubcategory.RightButton;
+                }
+                else
+                {
+                    touchEventArgs.eventSubCategory = (int) TouchEventSubcategory.SingleTouch;
+                }
+
+                TouchPanelProcessor.OnTouchedEvent(touchEventArgs.eventSubCategory, touchEventArgs.X1, touchEventArgs.Y1);
+
+            }
+
+            // Browse our samples repository: https://github.com/nanoframework/samples
+            // Check our documentation online: https://docs.nanoframework.net/
+            // Join our lively Discord community: https://discord.gg/gCyBu8T
+        }
+
+    }
+    public class TouchEventArgs : EventArgs
```

Comment:
All classes and enums should be in their own file. And should have the /// comments on all their public members. This is as well for readability and clarity in the code. Helping to use it as the /// comments will show up with intellisence.

Reply:
I agree with the separation of enum, classes etc in their own files for a fully professional project example...but that means that i have to create a completely new sample PROJECT and not one single file....is that how you go ahead for sample programs that are supposed to be targeted specially for newbies to jump start developing on the NF?

Reply:
If you are using Visual Studio, right click on the class and enum and click move to a new file. It will just love as magic.
And the quality of the samples and the code even for samples is what shows the professionalism. Again, I can fully understand the frustration it can generate at the beginning. I went there as well! And then after some efforts, everything is very natural and will be easier.